### PR TITLE
BUG: Runge-Kutta wrong TimeBounds implementation

### PR DESCRIPTION
**Pull request by josempozo**
The aimed normalization of the time apparently messed up the Runge-Kutta 
implementation resulting in a wrong algorithm. The implementation has been corrected to remove this bug. 

The existing test (itkTimeVaryingVelocityFieldIntegrationImageFilterTest) only tested for the integration of a homogeneous and constant velocity field, where the error could not be detected.
I have extended the test adding the case of a non-trivial (inhomogeneous and time-variaying) velocity field. This test is not passed by the previous implementation.

Added boolean flag to control the interpretation of LowerTimeBound and UpperTimeBound as rates 
in the velocity field time span. If true, it is equivalent to consider that the velocity field is defined in the normalized time interval [0, 1], ignoring the input origin and spacing in the temporal dimension.

This normalization was the previous behaviour. Thus, the default is true for backwards compatibility.

### CODE REVIEW

File: Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx [hunk 1]
```diff
@@ -64,7 +64,7 @@ itkTimeVaryingVelocityFieldIntegrationImageFilterTest(int, char *[])
   index.Fill(0);
   VectorType displacement;
 
-  auto = IntegratorType::New();
+  auto inverseIntegrator = IntegratorType::New();
```
Comment (dzenanz) [inline Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:67]: This commit should be squashed with the one which introduced the use of `auto` here.

File: Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx [hunk 2]
```diff
@@ -173,6 +173,10 @@ itkTimeVaryingVelocityFieldIntegrationImageFilterTest(int, char *[])
   integrator->SetInput(timeVaryingVelocityField);
   integrator->SetLowerTimeBound(0.2);
   integrator->SetUpperTimeBound(0.8);
+  /* This time bounds are meant to be absolute
```
Comment (dzenanz) [inline Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:176]: This commit should be squashed with the one which introduced this option.

File: Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx [hunk 3]
```diff
@@ -123,7 +123,7 @@ itkTimeVaryingVelocityFieldIntegrationImageFilterTest(int, char *[])
   size[0] = 3;
   size[1] = 3;
   size[2] = 401;
-  size[3] = 41;
+  size[3] = 61;
```
Comment (dzenanz) [inline Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:126]: Commit message should be updated to say "This a a modification to the time-varying test".

File: Modules/Filtering/DisplacementField/include/itkTimeVaryingVelocityFieldIntegrationImageFilter.h [hunk 4]
```diff
@@ -144,6 +144,18 @@ class ITK_TEMPLATE_EXPORT TimeVaryingVelocityFieldIntegrationImageFilter
    */
   itkGetConstMacro(NumberOfIntegrationSteps, unsigned int);
 
+  /**
```
Comment (dzenanz) [inline Modules/Filtering/DisplacementField/include/itkTimeVaryingVelocityFieldIntegrationImageFilter.h:147]: This commit should be squashed with the previous one.

File: Modules/Filtering/DisplacementField/include/itkTimeVaryingVelocityFieldIntegrationImageFilter.hxx [hunk 5]
```diff
@@ -18,8 +18,6 @@
 #ifndef itkTimeVaryingVelocityFieldIntegrationImageFilter_hxx
 #define itkTimeVaryingVelocityFieldIntegrationImageFilter_hxx
 
-#include "itkTimeVaryingVelocityFieldIntegrationImageFilter.h"
```
Comment (dzenanz) [inline Modules/Filtering/DisplacementField/include/itkTimeVaryingVelocityFieldIntegrationImageFilter.hxx]: This should be squashed with the first commit.

File: Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx [hunk 6]
```diff
@@ -99,5 +102,113 @@ itkTimeVaryingVelocityFieldIntegrationImageFilterTest(int, char *[])
     return EXIT_FAILURE;
   }
 
+
+  /* Second test with inhomogeneous and time-varying velocity field
```
Comment (dzenanz) [inline Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:106]: This is an awesome improvement to the test!

**Review by dzenanz — APPROVED** [review]
Looks great. Some commits should be squashed. Thank you for this contribution.

File: Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx (hunk 3 above)
Comment (josempozo) [reply Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:126]: I'm not sure how to change this message now.

But I could merge the commit with the previous one. Do you think this is a good idea?

File: Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx (hunk 6 above)
Comment (josempozo) [reply Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:106]: Thanks.

File: Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx (hunk 3 above)
Comment (dzenanz) [reply Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:126]: Here are some explanations of how to change a commit message:
https://stackoverflow.com/questions/1884474/change-old-commit-message-using-git-rebase
If you use some GUI tool such as TortoiseGit, it is even easier.

It is better not to merge, as making time normalization optional, and exercising that option are two different things, and it is best to have them as separate commits.

Comment (josempozo) [reply Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:126]: Thanks for the pointer. I think that I have resolved it now. Let me know if it looks correct from your side.

Comment (dzenanz) [reply Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:126]: You had two branches (one with original message, one with updated message) which were merged at the end, producing a strange looking history when linarized by GitHub. I picked the correct one and pushed it your branch. It is ready for merging now. Let's wait until CI finishes before doing that.

Comment (josempozo) [reply Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx:126]: Thanks for solving that merge.

**Pull request merged**